File: IR LLVM/Micro09/Micro09.c
#include <stdlib.h>
#include <stdio.h>

int main(){
    float preco;
    float venda;
    float novoPreco;

    printf("Digite o preco: ");
    scanf("%f", &preco);
    printf("Digite o a venda: ");
    scanf("%f", &venda);

    if((venda > 500) || (preco < 30)){
        novoPreco = preco + ((10.0/100.0) * 100.0);
    }
    else{
        if(((venda >= 500) && (venda <= 1200)) || ((preco >= 30) && (preco <= 80))){
            novoPreco = preco - ((20.0 / 100.0) * 100.0);
        }
    }

    printf("O novo preco é %f\n", novoPreco);
}

File: PASCAL/Micro09.pas
program micro09;

var 
preco, venda, novo_preco: real;

begin
    write('Digite o preco: ');
    readln(preco);
    write('Digite a venda: ');
    readln(venda);

    if (venda > 500) or (preco < 30) then
    begin
        novo_preco := preco + ((10 / 100) * 100);
    end
    else
    begin
        if ((venda >= 500) and (venda <= 1200)) or ((preco >= 30) and (preco < 80)) then
        begin
            novo_preco := preco - ((20 / 100) * 100);
        end;
    end;
    write('O novo preco e, ');
    writeln(novo_preco);
end.